Assets/Scripts/UI/Tooltip.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                if (textResized)
+                {
+                    cameraDistance = Vector3.Distance(Camera.main.transform.position, this.transform.position);
+                    tooltipStyle.fontSize = tooltipStyleShadow.fontSize = Mathf.RoundToInt(tooltipSize - (cameraDistance / 3));
+                }
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x - 40, Event.current.mousePosition.y + tooltipPosition.y, 100f, 20f), image, tooltipStyle);
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x - tooltipShadowPosition.x, Event.current.mousePosition.y + tooltipPosition.y - tooltipShadowPosition.y, 100f, 20f), currentText, tooltipStyleShadow);
+                GUI.Label(new Rect(Event.current.mousePosition.x + tooltipPosition.x, Event.current.mousePosition.y + tooltipPosition.y, 100f, 20f), currentText, tooltipStyle);
+                if (tooltipUIPanel != null)
+                {
+                    tooltipUIPanel.transform.localPosition = new Vector3(Event.current.mousePosition.x + tooltipPosition.x - Screen.width / 2f, -Event.current.mousePosition.y + tooltipPosition.y + Screen.height / 2f, 0);
+                }
+            }
+            // Display of text/tooltip that fixed to the object
+            else if (showTooltip && fixedToTheObject && over)
+            {
+                positionToScreen = Camera.main.WorldToScreenPoint(transform.position);
+                cameraDistance = Vector3.Distance(Camera.main.transform.position, this.transform.position);
+                if (textResized)
+                {
+                    tooltipStyle.fontSize = tooltipStyleShadow.fontSize = Mathf.RoundToInt(tooltipSize - (cameraDistance / 3));
+                }
+                //GUI.DrawTexture(new Rect((positionToScreen.x + tooltipPosition.x) - 40f, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), image, ScaleMode.ScaleToFit);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x - 40, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), image, tooltipStyle);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x - tooltipShadowPosition.x, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10) - tooltipShadowPosition.y, 100f, 20f), currentText, tooltipStyleShadow);
+                GUI.Label(new Rect(positionToScreen.x + tooltipPosition.x, -positionToScreen.y + Screen.height + tooltipPosition.y / (cameraDistance / 10), 100f, 20f), currentText, tooltipStyle);
+                if (tooltipUIPanel != null)
+                {
+                    tooltipUIPanel.transform.localPosition = new Vector3(positionToScreen.x + tooltipPosition.x - Screen.width / 2f, positionToScreen.y - Screen.height / 2f + tooltipPosition.y / (cameraDistance / 10), positionToScreen.z);
+                }
```

Comment:
Before I go into detail in this just a basic question. It seems like you creating everything in Code. Can't you do the same thing with a prefab and in the inspector?

Reply:
As far as i found, the ways of making tooltip are mostly manually creating a Canvas, UI Panel and attach to each objects. I just don't like it at all since we have so much more things to take care of. So drawing a Tooltip by GUI Label or GUI Texture is in my opinion the best way to do it.

The code you highlighted is for positioning the Tooltip Panel if someone can design one, so the Tooltip Panel will show first and have the Text on top. Well, i don't have the talent so will just leave it there for now XD

Reply:
Hmm if you don't dynamically change something with the Canvas, UI Panel and the GUI stuff, you should be able to create a prefab for it as well. But I can have a look into that as well. Just make sure to create an **issue** for it!